src/core/border_router/routing_manager.cpp

```diff
@@ -1234,6 +1238,10 @@ void RoutingManager::DiscoveredPrefixTable::ProcessDefaultRoute(const Ip6::Nd::R
     Entry      *entry;
     Ip6::Prefix prefix;
 
+    aRouter.mManagedAddressFlag = aRaHeader.IsManagedAddressFlagSet();
```

Comment:
nit: we should consider renaming `ProcessDefaultRoute` to `ProcessRaHeader` since the router may not be a default router.

Reply:
Good point! renamed it in new push.